rust-runtime/aws-smithy-http-server/src/plugin/layer.rs

```diff
@@ -23,6 +23,12 @@ where
     }
 }
 
+// Without more information about what the layer `L` does, we can't know whether it's appropriate
```

Comment:
Not too happy about this. With `PluginLayer` and LayerPlugin`, a user can effectively circumvent the whole point of this PR by converting an e.g. model plugin to a layer and using `LayerPlugin` to get a HTTP plugin.

Reply:
It seems a bit heavy-handed to force them to use a collection of plugins to make use of a _single_ plugin that they construct via a layer.

I think it's ok we go ahead with this. I can't think of valid reasons for why someone might want to do a plugin -> layer -> plugin conversion.